### ISSUE TITLE
Release v1.29.0 is having App version 1.28

### ISSUE THREAD

**Issue by sathakatheef** [kubernetes-sigs/aws-ebs-csi-driver#1981]
Hi Team
It seems the release v1.29.0 that was out today is having app version 1.28.0 on the chart [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/v1.29.0/charts/aws-ebs-csi-driver/Chart.yaml#L2). Is this expected? Please clarify. Thanks.

**PS**: The branch **release-1.29** is having the app version 1.29.0 on the chart [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/release-1.29/charts/aws-ebs-csi-driver/Chart.yaml#L2) which also belongs to today's change. So i don't think today's [release](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/releases/tag/v1.29.0) of v1.29.0 is off this branch.

/triage support

**Comment by torredil** [comment]
Hi @sathakatheef 

Helm releases are only performed when `Chart.yaml` is modified on the master branch: https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/ad393388f4e5e5db818e419dfbb49caa9ac61838/.github/workflows/helm-chart-release.yaml#L6

The latest Helm release correctly points to driver image `1.29.0`:

```
$ helm repo add aws-ebs-csi-driver https://kubernetes-sigs.github.io/aws-ebs-csi-driver
$ helm repo update

$ helm upgrade --install aws-ebs-csi-driver \
    --namespace kube-system \
    aws-ebs-csi-driver/aws-ebs-csi-driver

$ kubectl describe pod ebs-csi-controller-cd4d4748b-88qzh -n kube-system | grep "Image"
    Image:         public.ecr.aws/ebs-csi-driver/aws-ebs-csi-driver:v1.29.0
``` 

/close

**Comment by sathakatheef** [comment]
Hi @torredil 
Yup you are correct that the master branch points to the driver image 1.29.0. 

But i would like to install the latest release v1.29.0 in my cluster in which the app version is pointing to 1.28.0 which will be picked up by the image spec in the manifest/helm-chart [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/v1.29.0/charts/aws-ebs-csi-driver/templates/controller.yaml#L73) (as in below screenshot - downloaded the source code from the latest release [here](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/releases/tag/v1.29.0)).
<img width="1000" alt="image" src="https://github.com/kubernetes-sigs/aws-ebs-csi-driver/assets/52728017/cab8eb8e-a1af-41cd-a261-12292add0037">.

So i am pointing to that, Release v1.29.0 is having the driver image of v1.28.0.

**Comment by ConnorJC3** [comment]
@sathakatheef downloading the source code zip isn't a supported installation method and isn't expected to work.

Please install via our Helm repository `https://kubernetes-sigs.github.io/aws-ebs-csi-driver`, more information is available [in the Helm section of our installation docs](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/blob/master/docs/install.md#helm)

**Comment by AndrewSirenko** [comment]
@sathakatheef as a quirk of our release process that @torredil mentioned, the v1.29.0 tag commit does not have the v2.29.0 helm chart. The v1.29.0 tag denotes that the driver code itself was what was released as v1.29 of our add-on. 

See [Possible wrong tagged commit for `v1.25.0` · Issue #1839 · kubernetes-sigs/aws-ebs-csi-driver](https://github.com/kubernetes-sigs/aws-ebs-csi-driver/issues/1839)

**Comment by sathakatheef** [comment]
Great. Thanks for the clarification 👍.
@AndrewSirenko @torredil @ConnorJC3
